jgrapht-core/src/main/java/org/jgrapht/alg/BergeGraphInspector.java

```diff
@@ -0,0 +1,1019 @@
+ * <p>Special Thanks to Maria Chudnovsky for her kind help.
+ * 
+ * <p>The runtime complexity of this implementation is $O(|V|^9|)$. This implementation is far more efficient than 
+ * simplistically testing whether graph  $G$ or its complement $\overline{G}$ have an odd cycle, because testing 
+ * whether one graph can be found as an induced subgraph of another is 
+ * <a href="https://en.wikipedia.org/wiki/Induced_subgraph_isomorphism_problem">known</a> to be NP-hard.
+ * 
+ * @author Philipp S. Kaesgen (pkaesgen@freenet.de)
+ * @since 2018
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class BergeGraphInspector<V,E>{
+    /**
+     * Checks whether two paths which both intersect in the vertex m have another common vertex
+     * @param g A Graph
+     * @param s1 A Vertex
+     * @param s2 A Vertex
+     * @param m A Vertex
+     * @return whether there is an edge except for m
+     */
+    protected boolean haveNoEdgeDisregardingM(Graph<V,E> g,GraphPath<V,E> s1, GraphPath<V,E> s2,V m){
+        for (V v1 : s1.getVertexList())
+            if (v1 != m)
+                for (V v2 : s2.getVertexList())
+                    if (v2 != m&&!g.containsEdge(v1,v2))
+                        return true;
+                        
+        return false;
+    }
+    
+    
+    /**
+     * Lists the vertices which are covered by two paths
+     * @param g A Graph
+     * @param p1 A Path in g
+     * @param p2 A Path in g
+     * @return Set of vertices covered by both p1 and p2 
+     */
+    protected Set<V> intersectGraphPaths(Graph<V,E> g,GraphPath<V,E> p1, GraphPath<V,E> p2){
+        Set<V> res = new HashSet<V>();
+        res.addAll(p1.getVertexList());
+        res.retainAll(p2.getVertexList());
+        return res;
+    }
+    
+    /**
+     * Assembles a GraphPath of the Paths S and T. Required for the Pyramid Checker
+     * @param g A Graph
+     * @param S A Path in g
+     * @param T A Path in g
+     * @param M A set of vertices in g
+     * @param m A vertex
+     * @param b1 A base vertex
+     * @param b2 A base vertex
+     * @param b3 A base vertex
+     * @param s1 A vertex
+     * @param s2 A vertex
+     * @param s3 A vertex
+     * @return The conjunct path of S and T
+     */
+    protected GraphPath<V,E> P(Graph<V,E> g,GraphPath<V,E> S, GraphPath<V,E> T, Set<V> M, V m, V b1, V b2, V b3, V s1, V s2, V s3){
+        if (S!=null&&T!=null){
+            if (s1==b1){
+                if (b1==m){
+                    List<E> edgeList = new LinkedList<E>();
+                    edgeList.add(g.getEdge(s1, b1));
+                    return new GraphWalk<V,E>(g,s1,b1,edgeList,g.getEdgeWeight(edgeList.get(0)));
+                }
+                else{
+                    return null;
+                }
+            }
+            else{
```

Comment:
Since all the previous code paths are returning from the method, you can skip the else here. Similar comment applies later on in this function.